src/test/java/com/bobocode/hoverla/bring/context/ApplicationContextImplTest.java

```diff
@@ -0,0 +1,157 @@
+
+import com.bobocode.hoverla.bring.exception.NoSuchBeanException;
+import com.bobocode.hoverla.bring.exception.NoUniqueBeanException;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.mockito.Mockito;
+
+import java.math.BigDecimal;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertInstanceOf;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.any;
+import static org.mockito.Mockito.anyList;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@TestInstance(value = TestInstance.Lifecycle.PER_CLASS)
+public class ApplicationContextImplTest {
+
+    private ApplicationContext applicationContext;
+    private BeanScanner beanScannerOne;
+    private BeanScanner beanScannerTwo;
+    private BeanValidator validator;
+    private BeanInitializer initializer;
+
+    @BeforeAll
+    void init() {
+        beanScannerOne = Mockito.mock(BeanScanner.class);
+        beanScannerTwo = Mockito.mock(BeanScanner.class);
+        List<BeanScanner> beanScannerList = Arrays.asList(beanScannerOne, beanScannerTwo);
+
+        validator = Mockito.mock(BeanValidator.class);
+
+        initializer = Mockito.mock(BeanInitializer.class);
+
+        BeanDefinition beanDefinitionStringOne = Mockito.mock(BeanDefinition.class);
+        BeanDefinition beanDefinitionStringTwo = Mockito.mock(BeanDefinition.class);
+        BeanDefinition beanDefinitionInteger = Mockito.mock(BeanDefinition.class);
+
+        when(beanScannerOne.scan()).thenReturn(Arrays.asList(beanDefinitionStringOne, beanDefinitionInteger));
+        when(beanScannerTwo.scan()).thenReturn(Collections.singletonList(beanDefinitionStringTwo));
+
+        when(beanDefinitionStringOne.name()).thenReturn("String");
+        doReturn(String.class).when(beanDefinitionStringOne).type();
+        when(beanDefinitionStringOne.instance()).thenReturn("String bean");
+
+        when(beanDefinitionInteger.name()).thenReturn("Integer");
+        doReturn(Integer.class).when(beanDefinitionInteger).type();
+        when(beanDefinitionInteger.instance()).thenReturn(2);
+
+        when(beanDefinitionStringTwo.name()).thenReturn("One more string bean");
+        doReturn(String.class).when(beanDefinitionStringTwo).type();
+        when(beanDefinitionStringTwo.instance()).thenReturn("String bean");
+
+        applicationContext = new ApplicationContextImpl(beanScannerList, validator, initializer);
+    }
+
+    @Test
+    @DisplayName("Scan, validate and initialized methods have been called")
```

Comment:
(nit) I'd say "Calls bean scanners, context validator and bean initializer during on creation"